BryantUniversity/BryantUniversity/ViewModels/SemesterDetailsViewModel.cs

```diff
@@ -0,0 +1,23 @@
+﻿using BryantUniversity.Data;
+using BryantUniversity.Models;
+using BryantUniversity.Models.Repo;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using System.Web.Mvc;
+
+namespace BryantUniversity.ViewModels
+{
+    public class SemesterDetailsViewModel
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public string UrlSlug { get; set; }
+
+        public IList<SemesterDetails> semesterDetails = new List<SemesterDetails>();
```

Comment:
Is this supposed to be declared as an IList or a List?

Reply:
Well IList is an interface rather than a regular class such as List, so you actually can't do something like public IList<SemesterDetails> semesterDetails = new IList<SemesterDetails>();

Reply:
I understand that IList is an interface. I was just wondering why you're using IList instead of List. I'm guessing that it's to allow for flexibility, but correct me if I'm wrong.

Reply:
Well the CalendarRepo returns an IList so if its declared using List instead of IList then we get an error on Line 18 of HomeController stating:

Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0266	Cannot implicitly convert type 'System.Collections.Generic.IList<BryantUniversity.Models.SemesterDetails>' to 'System.Collections.Generic.List<BryantUniversity.Models.SemesterDetails>'. An explicit conversion exists (are you missing a cast?)	BryantUniversity	C:\Users\Sarthak\Documents\BryantUniversity\BryantUniversity\BryantUniversity\Controllers\HomeController.cs	18	Active

In addition, it does provide more flexibility.